packages/flutter/lib/src/widgets/binding.dart

```diff
@@ -303,6 +303,28 @@ mixin WidgetsBinding on BindingBase, SchedulerBinding, GestureBinding, RendererB
     });
 
     assert(() {
+      registerSignalServiceExtension(
+        name: 'lifecycleSuspend',
+        callback: () async {
+          // Let the engine know to stop scheduling frames.
+          await SystemChannels.lifecycle.send('AppLifecycleState.suspending');
+          // Let the framework know we're suspended (the message doesn't get echoed
+          // back from the engine).
+          handleAppLifecycleStateChanged(AppLifecycleState.suspending);
```

Comment:
We should probably document on `suspending` and `resumed` that these will also fire during hot reload?

Along the same lines: Does overloading this have any unintended side effects? What if an app developer is listing to these to do something, do we want that to happen on hot reload as well?

Reply:
I don't think this is the right approach, it is totally reasonable to have business logic tied to the app lifecycle state.